libass/ass_utils.h

```diff
@@ -30,6 +30,7 @@
 #include <math.h>
 
 #include "ass.h"
+#include "wyhash.h"
```

Comment:
The file is a verbatim copy of the upstream/official wyhash code. Of course, its licence permits us to change it however we see fit, but personally, I prefer to keep such code intact (unless they have bugs that need fixing), e. g. so that it can be easily updated when upstream code changes.

Reply:
Yes, I'm usually in favor of keeping external dependencies intact too. But this specific file looks like code written by inexperienced student and it's quite hard to understand what's going on in that. So I prefer to borrow algorithm only and not the source code.

I'm planning to check original upstream repo and try to simplify code and maybe even improve performance.

Reply:
As upstream appears to be active and even more so with MrSmile planning to submit improvements to upstream, using the header verbatim for now seems imho like the better option.

Reply:
Sadly, upstream has research code instead of production (all that `WYHASH_CONDOM`, `WYHASH_32BIT_MUM`, bunch of auxiliary functions) so that is unlikely to happen. I've done some experiments on wyhash [here](https://github.com/MrSmile/wyhash/tree/test) and it's indeed one of the fastest on bulk data. Reducing the number of multiply operations per byte can improve bulk performance, but I've not found a way to do so without sacrificing hash quality. Performance on small fixed-size elements can be improved but it's not the problem case.